docs/concepts/tech-stack.md

```diff
@@ -0,0 +1,132 @@
+# Tech Stack
+
+## Overview
+
+Canister authors can opt in to display the tech stack of the canister.
+
+Providing a standard format of such information makes it easier to build tools like a Canister Explorer.
+
+## JSON schema
+
+`tech_stack` has 5 top optional categories.
+
+- cdk
+- language
+- lib
+- tool
+- other
+
+Each category contains an Array of tech stack items.
+
+Each tech stack item must have a `"name"` field and can optionally define some custom fields, e.g. `"version"`.
+
+### Example
+
+```json
+{
+  "tech_stack": {
+    "language": [
+      {
+        "name": "rust",
+        "version": "1.75.0"
+      }
+    ],
+    "cdk": [
+      {
+        "name": "ic-cdk",
+        "version": "0.13.0"
+      }
+    ],
+    "lib": [
+      {
+        "name": "ic-cdk-timers"
+      },
+      {
+        "name": "ic-stable-structures"
+      }
+    ],
+    "other": [
+      {
+        "name": "bitcoin"
+      }
+    ],
+    "tool": [
+      {
+        "name": "dfx"
+      }
+    ]
+  }
+}
+```
+
+## How to set it in `dfx.json`
+
+The example above was generated from the `dfx.json` configuration below.
+
+```json
+{
+  "type": "motoko",
+  "main": "main.mo",
+  "tech_stack": {
+    "cdk": [
+      {
+        "name": "ic-cdk",
+        "custom_fields": [
```

Comment:
I think we should give the team some time to review the structure of this data

Reply:
I get a bit lost with the fact that `version` down in the custom_fields array turns up as a sibling to `name`

Reply:
I added a custom field to the `bitcoin` item: `"address": "bcrt1qfe264m0ycx2vcqvqyhs0gpxk6tw8ug6hqeps2d"`.

`"version"` and `"address"` are both custom fields.

I hope this another example beyond `"version"` can make the intention more clear.